### PR TITLE
New branch for login networking

### PR DESCRIPTION
**Pull request by YKo20010**

### CODE REVIEW

**Review by kevinchan159 — APPROVED** [review]
looks good

File: acai-ios/Extensions/String+Extension.swift [hunk 1]
```diff
@@ -40,14 +40,19 @@ extension String {
...
     }
 
-    // TODO: update name requirements
     func isValidName() -> Bool {
-        return self.count > 0
+        let nameRE = "[a-zA-Z]+"
+        return NSPredicate(format: "SELF MATCHES %@", nameRE).evaluate(with: self)
     }
 
     // TODO: update password requirements
```
Comment (cornell-appdev) [inline acai-ios/Extensions/String+Extension.swift:48]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/cuappdev/acai-ios/blob/72216c842d4a8a7900732ab1d084df62b0c91d31/acai-ios/Extensions/String+Extension.swift#L48"></span>TODOs should be resolved (update password requirements).<br />
<code>todo</code> <code>String+Extension.swift:48</code></td>
    </tr>
  </tbody>
</table>

**Comment by cornell-appdev** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR, try to keep changes smaller if you can</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Skipping duplicate build file in Compile Sources build phase: /Users/vagrant/git/acai-ios/Models/MenuItem.swift (in target ‘acai-ios’)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Skipping duplicate build file in Compile Sources build phase: /Users/vagrant/git/acai-ios/Extensions/UIFont+Shared.swift (in target ‘acai-ios’)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint not installed, download from https://github.com/realm/SwiftLint</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/cuappdev/acai-ios/blob/72216c842d4a8a7900732ab1d084df62b0c91d31/LoginViewController.swift#L181">LoginViewController.swift#L181</a> - <strong><a href="https://github.com/cuappdev/acai-ios/blob/72216c842d4a8a7900732ab1d084df62b0c91d31/acai-ios/ViewControllers/LoginViewController.swift#L181">acai-ios/ViewControllers/LoginViewController.swift#L181</a></strong>: variable ‘self’ was written to, but never read  <br /><code>
            createUser(email: email, firstName: firstName, lastName: lastName, password: password, phoneNumber: phoneNumber).observe { [weak self] result in
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/cuappdev/acai-ios/blob/72216c842d4a8a7900732ab1d084df62b0c91d31/LoginViewController.swift#L199">LoginViewController.swift#L199</a> - <strong><a href="https://github.com/cuappdev/acai-ios/blob/72216c842d4a8a7900732ab1d084df62b0c91d31/acai-ios/ViewControllers/LoginViewController.swift#L199">acai-ios/ViewControllers/LoginViewController.swift#L199</a></strong>: variable ‘self’ was written to, but never read  <br /><code>
            getUser(email: email, password: password).observe { [weak self] result in
</code></td>
    </tr>
  </tbody>
</table>

### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
String+Extension.swift | 48 | TODOs should be resolved (update password requirements). (todo)


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by doctor3w — APPROVED** [review]
looks great! Thank you for going back and doing all that, didn't know it would be such a pain.

**Pull request merged**